test_private_network.py

```diff
@@ -319,3 +319,40 @@ def test_private_network_only_on_all_images(prober, create_server, image):
     # Make sure that no public interface is configured
     public_addresses = server.configured_ip_addresses(is_global=True)
     assert len(public_addresses) == 0
+
+
+def test_private_network_attach_later(server, private_network):
+    """Private network ports can be attached to an already running server.
+
+    """
+
+    subnet = private_network.add_subnet('10.0.0.0/24')
+
+    # Assert no eth1 interface is present
+    assert not server.host.interface('eth1').exists
```

Comment:
Is this check necessary? If a private network is already present, the check further down would find two private interfaces and fail anyway.